### PR TITLE
Serialize modal popups at startup

### LINKED ISSUE #1988: Check we're not presenting multiple modal dialogs at once on startup
The sync on startup, add-on update notification, and Anki update notification dialogs are all modal. We should make sure we don't present more than one at once, perhaps by only triggering the update checks after an initial sync, and only one after the other.

https://forums.ankiweb.net/t/dialogue-box-collision-on-startup/21724

### PR DESCRIPTION
**Pull request by sam1penny**
Fixes #1988 by calling `maybe_check_for_addon_updates()` after an initial sync

### CODE REVIEW

**Comment by dae** [comment]
Thanks Sam! I'm not sure we can close #1988 yet though, as the "anki updated" popup still needs to be delayed until the others have run (or vice versa). Would you like to try handle that too in this PR?

**Comment by sam1penny** [comment]
> Thanks Sam! I'm not sure we can close #1988 yet though, as the "anki updated" popup still needs to be delayed until the others have run (or vice versa). Would you like to try handle that too in this PR?

Sure - how about like this?

I might've overcomplicated it a bit by making it as general as possible - e.g providing an optional `on_done` argument to `maybe_check_for_addon_updates` even though it's only called once. Let me know if you want me to make any changes :)

File: qt/aqt/main.py [hunk 1]
```diff
@@ -893,17 +892,23 @@ def setupAddons(self, args: list | None) -> None:
...
+            if on_done:
+                on_done(log)
+            self.on_updates_installed(log)
+
         if elap > 86_400 or self.pm.last_run_version() != point_version():
             check_and_prompt_for_updates(
                 self,
                 self.addonManager,
-                self.on_updates_installed,
+                wrap_on_updates_installed,
```
Comment (dae) [inline qt/aqt/main.py:911]: This skips calling on_done when time hasn't elapsed. Not a big deal in the grand scheme of things, but it's inconsistent with the auto sync code that always calls the callable.

**Review by dae — COMMENTED** [review]
on_done seems like a reasonable approach to me :-)

**Comment by sam1penny** [comment]
Thanks for the review :)

**Comment by dae** [comment]
Thanks Sam!

**Pull request merged**